File: Palindromo/palindromo.java

package palindromo;

import java.util.Scanner;

public class Palindromo {

    public static void main(String[] args) {
        Scanner T = new Scanner(System.in);
        
        System.out.println("Ingrese un texto:");
        String palabra = T.nextLine();
        
        
        if(es_palindromo(palabra)){
            System.out.println("La palabra "+palabra+" es un palindromo");
        }else{
            System.out.println("La palabra "+palabra+" no es un palindromo");
        }
        
    }
    
    public static boolean es_palindromo(String palabra){
        String palabraI = invertir_cadena(palabra);
   
        return palabra.equals(palabraI);
    }
    
    public static String invertir_cadena(String cadena){
        String cadenaI="";
        
        for(int i = cadena.length()-1; i>=0; i--){
            cadenaI += cadena.charAt(i);
        }
        
        return cadenaI;
    }
    
}

File: Palindromo/palindromo.cpp
#include <iostream>
using namespace std;

bool es_palindromo(const string&);
string invertir_cadena(string);

int main() {
    string palabra;
    cout << "Bienvenido a Palindromo" << endl;
    cout << "Ingresa una palabra" << endl;
    cin >> palabra;
    
    if(es_palindromo(palabra)){
        cout << "La palabra es palindromo"<<endl;
    }else{
        cout << "La palabra NO es palindromo"<<endl;
    }

    return 0;
}

bool es_palindromo(const string& palabra) {
    string palabraI = invertir_cadena(palabra);

    if(palabra==palabraI){
        return true;
    }else{
        return false;
    }
}

string invertir_cadena(string cadena){
    string cadenaI="";

    for(int i = cadena.length() - 1; i >= 0; i--){
        cadenaI += cadena[i];
    }

    return cadenaI;

}



